src/screens/SignInScreen/index.tsx

```diff
@@ -160,11 +196,19 @@ class SignIn extends React.Component<Props, SignInState> {
 
     private handleSignIn = () => {
         const { email, password } = this.state;
+        const { configs, captcha_response } = this.props;
 
-        this.props.signIn({
-            email,
-            password,
-        });
+        switch (configs.captcha_type) {
+            case 'recaptcha':
+            case 'geetest':
+                this.props.signIn({ email, password, captcha_response });
+                break;
+            default:
+                this.props.signIn({ email, password });
+                break;
+        }
+
+        this.props.resetCaptchaState();
```

Comment:
this.props.resetCaptchaState() should be called:
- if we receive sign in error or sign in success, because if 2FA is enabled, you reset captcha's state before it's sent in request; (componentWillReceiveProps or it's analog)
- componentWillUnmount()